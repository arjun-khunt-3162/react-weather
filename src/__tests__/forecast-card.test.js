import React from 'react';
import { render, screen, waitForLoadingToFinish } from '../test/app-test-utils';
import ForecastCard from '../components/forecast-card';

const testProps = {
  location: 'Eldoret',
  units: 'metric',
};

describe('ForecastCard', () => {
  test('renders the weekly forecast for the specified location', async () => {
    renderForecast(testProps);

    await waitForLoadingToFinish();

    const forecast = screen.getAllByRole('listitem').map((listItem) => {
      return listItem.textContent;
    });

    expect(forecast).toMatchSnapshot(`
      Array [
        "Saturday22° / 22°",
        "Sunday22° / 22°",
        "Monday22° / 22°",
        "Tuesday20° / 20°",
        "Wednesday21° / 21°",
      ]
    `);
  });
});

function renderForecast(testProps) {
  render(<ForecastCard {...testProps} />);
}
