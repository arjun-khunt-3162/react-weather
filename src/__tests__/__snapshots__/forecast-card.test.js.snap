// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastCard renders the weekly forecast for the specified location: 
      Array [
        "Saturday22° / 22°",
        "Sunday22° / 22°",
        "Monday22° / 22°",
        "Tuesday20° / 20°",
        "Wednesday21° / 21°",
      ]
     1`] = `
[
  "Saturday22° / 22°",
  "Sunday22° / 22°",
  "Monday22° / 22°",
  "Tuesday20° / 20°",
  "Wednesday21° / 21°",
]
`;
